historical_surveillance/views.py

```diff
@@ -0,0 +1,328 @@
+            school_category = form.cleaned_data['category_of_school']
+            # change the updated by column to automatically insert the name of the logged in user updating the column
+            school_updated = form.cleaned_data['updated_by']
+            if not School.objects.filter(School_Code=school_code,
+                                         created_by=school_created_by,
+                                         updated_by=school_updated,
+                                         School_Name=school_name,
+                                         District_Name=district_name,
+                                         category_of_school=school_category).exists():
+                form.save()
+                return redirect("/")
+
+            else:
+                error_message = "The record you are trying to create already exists."
+
+        stu = {
+            "name_of_school": data,
+            "error_message": error_message,
+            "form" : form,
+        }
+        return render(request, 'school.html', stu)
+    return render(request, 'school.html', {'form':form, 'name_of_school':data})
+
+
+# This function controls the enrollment by school,
+# grade and gender form view, processes the and submits form data to the database
+def enrollment(request):
+    form = EnrollmentForms()
+    district_df = pd.DataFrame(District.objects.all().values())
+    districts_names = district_df['District_Name']
+    return render(request, 'enrollment.html', {'form': form, 'district_names': districts_names})
+
+
+# This function controls the total enrollment and school capacity form, processes the form data and submits
+# it to the database
+def enroll_class(request):
+    form = AggregateEnrollmentForms()
+    if request.method == 'POST':
+        form = AggregateEnrollmentForms(request.POST)
+        if form.is_valid():
+            school_form = form.cleaned_data['name_of_school']
+            academic_year_form = form.cleaned_data['academic_year']
+            district_form = form.cleaned_data['district_of_school']
+            if not AggregateEnrollment.objects.filter(name_of_school=school_form, academic_year=academic_year_form,
+                                                      district_of_school=district_form).exists():
+                form.save()
+                return redirect("/")
+        else:
+            messages.add_message(request, messages.INFO, 'Data has been entered previously')
+
+    return render(request, 'enroll_class.html', {'form': form})
+
+
+# This view controls the enrollment / Capacity views table
+def enrolled(request):
+    data = AggregateEnrollment.objects.all()
+
+    stu = {
+        "name_of_school": data
+    }
+    return render(request, 'enrolled.html', stu)
+
+#change this to get a form to select the district and pass it as a parameter to present filter the table and present
+# the data for each district
+def enrolled_trend_district_1(request):
```

Comment:
here we can get which district we are looking at in the second parameter if we set the URL correctly.